src/libutil/references.cc

```diff
@@ -29,15 +32,21 @@ static void search(
     });
 
     for (size_t i = 0; i + refLength <= s.size(); ) {
```

Comment:
```suggestion
    /* Nested loops, but the inner loop also advances i, so
       this is an asymptotically optimal solution for finding
       hash-like substrings. */
    for (size_t i = 0; i + refLength <= s.size(); ) {
```

It screams mutual recursion, but I'll ignore that.